packages/rest/src/rest.server.ts

```diff
@@ -167,6 +166,15 @@ export class RestServer extends Context implements Server, HttpServerLike {
     this._setupRequestHandler(options);
 
     this.bind(RestBindings.HANDLER).toDynamicValue(() => this.httpHandler);
+
+    this._httpServer = new HttpServer(
```

Comment:
We need to defer creation of HttpServer instance until `start` is called, so that we can pick any changes in HOST/PORT bindings.

An outline of a test case to write:

```ts
const server = new RestServer({port: 80});
server.bind(RestBindings.PORT).to(0);
await server.start();
// get the actual port number and verify it's not 80
```